Add 'Working with Relationships' guide to Models

kevinkucharczyk opened the pull request:
Closes #785 

I've added a new guide which aims to clarify how to create/update/remove relationships.

michaelrkn commented:
Thanks for the quick turnaround on this!

michaelrkn commented on source/models/working-with-relationships.md:
````diff
@@ -0,0 +1,116 @@
+```javascript
+let post = this.store.peekRecord('post', 1);
+let comment = this.store.createRecord('comment', {
+  // comment data
````
I think this is self-explanatory and doesn't require a comment.

michaelrkn commented on source/models/working-with-relationships.md:
````diff
@@ -0,0 +1,116 @@
+comment.save();
+```
+
+This will create a new `comment` record and save it to the server. Ember Data will also update the post to include our newly created comment in its `comments` relationship. We could have also linked the two records together by updating the post's `hasMany` relationship:
````
I think make "We could have also..." into a new paragraph.

michaelrkn commented on source/models/working-with-relationships.md:
````diff
@@ -0,0 +1,116 @@
+```javascript
+let post = this.store.peekRecord('post', 1);
+let comment = this.store.createRecord('comment', {
+  // comment data
````
I think this is self-explanatory and doesn't require a comment.

michaelrkn commented on source/models/working-with-relationships.md:
````diff
@@ -0,0 +1,116 @@
+comment.save();
+```
+
+We can also update the `hasMany` relationship by pushing a record into the relationship:
````
I suggest "Alternatively, we could update the..."

michaelrkn commented on source/models/working-with-relationships.md:
```diff
@@ -0,0 +1,116 @@
+
+### Removing Relationships
+
+To remove a `belongsTo` relationship, we can just set it to `null`, which will also remove it from the inverse:
```
I'd suggest "inverse" -> "`hasMany` side"

michaelrkn commented on source/models/working-with-relationships.md:
```diff
@@ -0,0 +1,116 @@
+
+### Relationships as Promises
+
+While working with relationships it is important to remember that they return promises. In the examples above we have assumed that the records we were working on were already loaded into the store, but this may not always be the case.
```
In the above examples, we aren't actually working with the returned data, so that they returned promises isn't really relevant. I think removing the last sentence would avoid confusion about when the user needs to be concerned about the asynchronous nature of the relationship.

michaelrkn commented:
This looks really good! I've added a couple comments.

michaelrkn commented on source/models/working-with-relationships.md:
```diff
@@ -0,0 +1,116 @@
+### Creating Records
```
These should be ##, not ### (and all below ones, too).

michaelrkn commented:
Looks good! One last comment from me. Then can you squash your commits?

Also, did you sort out your [questions](https://github.com/emberjs/guides/issues/785#issuecomment-143548037) on your own, or should we wait for a final review by an ED expert?

kevinkucharczyk commented:
Okay, I've added updates according to your comments and squashed my commits.

I figured out the questions on my own, since I didn't receive any replies here nor on Slack (which is understandable, since it was the weekend). The methods I propose do work, though the question still remains whether what I did is, in fact, best practice.

I believe it would be good to have someone from ED review my approach.

wecc commented:
LGTM
